### PR TITLE
Add CITATION.cff

### LINKED ISSUE #449: [FEATURE] CITATION.cff file
I just wanted to cite xgcm and got [this](https://zenodo.org/record/6097129/export/hx#.YjFMhy-B30o) from zenodo. Many of the names are not properly documented.

I would love to have proper names for the bibtex files, and think we can achieve this with the CITATION.cff file. 

I will include all contributors on this if they want. So @rabernat @timothyas @jdldeauna @andersy005 @TomNicholas @ocefpaf @jrbourbeau @raphaeldussin @dcherian @rcaneill @bradyrx @anirban89 @Swanson-Hysell @willirath @apatlpo @dhruvbalwada @navidcy @leouieda @yz3062 @roxyboy, if you would like to be included in the xgcm citation, could you please respond with your **full name + orcid** below? 

Cheers

### PR DESCRIPTION
**Pull request by jbusecke**
- [x] Closes #449 
 - [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Review by navidcy — APPROVED** [review]

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,60 @@
+cff-version: 1.2.0
+message: "If you use this software, please cite it as below."
+authors:
+- family-names: "Abernathey"
+  given-names: "Ryan P."
+  orcid: "https://orcid.org/0000-0001-5999-4917"
+- family-names: "Busecke"
+  given-names: "Julius J. M."
+  orcid: "https://orcid.org/0000-0001-8571-865X"
+# - timothyas
```
Comment (navidcy) [inline CITATION.cff:10]:
```suggestion
- family-names: "Smith"
   given-names: "Timothy Andrew"
   orcid: "https://orcid.org/0000-0003-4463-6126"
```

Comment (timothyas) [reply CITATION.cff:10]: Thanks!

Comment (jbusecke) [reply CITATION.cff:10]: Ah thanks for directly suggesting here!

**Comment by jbusecke** [comment]
The remaining error should have been fixed in #472. Ill merge this now

**Comment by dcherian** [comment]
Nice work Julius. I've done this before and it's not fun!

**Pull request merged**